sirepo/template/radia.py

```diff
@@ -1092,51 +1126,179 @@ def _save_kick_map_sdds(name, x_vals, y_vals, h_vals, v_vals, path):
     return path
 
 
+def _stemmed_geometry(o, beam_axis, height_axis):
+    w, h, b = _geom_directions(beam_axis, height_axis)
+    ctr = sirepo.util.split_comma_delimited_string(o.center, float)
+    sz = sirepo.util.split_comma_delimited_string(o.size, float)
+    return w, h, b, [numpy.sum(w * ctr), numpy.sum(h * ctr)], [numpy.sum(w * sz), numpy.sum(h * sz)]
+
+
 def _undulator_termination_name(index, term_type):
     return f'termination.{term_type}.{index}'
 
 
-def _update_cuboid(cuboid, **kwargs):
-    return _update_geom_obj(cuboid, delim_fields=PKDict(segments=[1, 1, 1]), **kwargs)
+def _update_cee(o, beam_axis, height_axis, **kwargs):
+    return _update_cee_points(_update_geom_obj(o, **kwargs), beam_axis, height_axis)
 
 
-def _update_ell(ell, beam_axis, height_axis, **kwargs):
-    ell = _update_geom_obj(ell, **kwargs)
-    _update_ell_points(ell, beam_axis, height_axis)
-    return ell
+def _update_cee_points(o, beam_axis, height_axis):
+    g = _update_stemmed_points(o, beam_axis, height_axis)
+    w, h, b, c, s = g.geom
+    ax1, ax2, ay1, ay2, sx1, sx2, sy1 = g.points
 
+    sy2 = sy1 + o.armHeight
 
-def _update_ell_points(ell, beam_axis, height_axis):
-    w, h, b = _geom_directions(beam_axis, height_axis)
-    ctr = sirepo.util.split_comma_delimited_string(ell.center, float)
-    sz = sirepo.util.split_comma_delimited_string(ell.size, float)
-    c = [numpy.sum(w * ctr), numpy.sum(h * ctr)]
-    s = [numpy.sum(w * sz), numpy.sum(h * sz)]
+    o.points = _update_extrusion_points(
+        [
+            [ax1, ay1], [ax2, ay1], [ax2, ay2],
+            [sx2, ay2], [sx2, sy2], [ax2, sy2], [ax2, sy1], [sx1, sy1],
```

Comment:
From the looks of it for each call to _update_extrusion_points we do the exact same work except this line is different. I think more objects should be passed to the function (ex pass the whole g and o) and have it do more pulling things apart.

Reply:
agree, see recent comment